src/Datadog.Trace/Configuration/JsonConfigurationSource.cs

```diff
@@ -85,7 +86,7 @@ string IConfigurationSource.GetString(string key)
         /// <returns>The value of the setting, or null if not found.</returns>
         bool? IConfigurationSource.GetBool(string key)
         {
-            return GetValue<bool?>(key);
+            return GetValue<string>(key)?.ToBoolean();
```

Comment:
Not sure about this one since JSON supports real Boolean values (and numbers, etc). This is why `JsonConfigurationSource` doesn't derive from `StringConfigurationSource`. That's only for sources that only support string values (e.g. environment variables, `web.config`).